packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/entity_type.rs

```diff
@@ -47,34 +47,37 @@ impl<C: AsClient> PostgresStore<C> {
                 .ontology_dependency_map
                 .insert(entity_type_id, current_resolve_depth);
             let entity_type = match dependency_status {
-                DependencyStatus::Unknown => {
-                    if let Some(entity_type) = subgraph.vertices.ontology.get(entity_type_id) {
-                        Some(entity_type.clone())
-                    } else {
-                        let entity_type = Read::<EntityTypeWithMetadata>::read_one(
-                            self,
-                            &Filter::for_ontology_type_edition_id(entity_type_id),
-                        )
-                        .await?;
-                        Some(
-                            subgraph
-                                .vertices
-                                .ontology
-                                .entry(entity_type_id.clone())
-                                .or_insert(OntologyVertex::EntityType(Box::new(entity_type)))
-                                .clone(),
-                        )
+                DependencyStatus::Unresolved => {
+                    match subgraph
+                        .vertices
+                        .ontology
+                        .raw_entry_mut()
+                        .from_key(entity_type_id)
+                    {
+                        RawEntryMut::Occupied(entry) => Some(entry.into_mut()),
+                        RawEntryMut::Vacant(entry) => {
+                            let entity_type = Read::<EntityTypeWithMetadata>::read_one(
+                                self,
+                                &Filter::<EntityType>::for_ontology_type_edition_id(entity_type_id),
+                            )
+                            .await?;
+                            Some(
+                                entry
+                                    .insert(
+                                        entity_type_id.clone(),
+                                        OntologyVertex::EntityType(Box::new(entity_type)),
+                                    )
+                                    .1,
+                            )
+                        }
                     }
                 }
-                DependencyStatus::DependenciesUnresolved => {
-                    subgraph.vertices.ontology.get(entity_type_id).cloned()
-                }
                 DependencyStatus::Resolved => None,
             };
 
-            if let Some(OntologyVertex::EntityType(entity_type)) = entity_type {
-                for property_type_ref in entity_type.inner().property_type_references() {
-                    if current_resolve_depth.constrains_properties_on.outgoing > 0 {
+            if let Some(OntologyVertex::EntityType(entity_type)) = entity_type.cloned() {
```

Comment:
Not asking for this PR, but is there a way we can get around these clones by making these records immutable somehow? It seems that this is caused by a mutable borrow to `subgraph` so I'm wondering if these records can be guaranteed to not change so their references aren't tied to the mutable borrow on `subgraph`.

Guessing it would not be straightforward and would involve some lifetime nonsense or a secondary collection to hold onto the records while the subgraph just holds references to them or something.

Reply:
That's not easily possible. The subgraph is mutably borrowed in all `traverse_*` functions to be extended by the new vertices. One possible solution is to store all references (`.property_type_references()`, `.data_type_references()` etc.) before, these are only called conditionally. I will do an attempt to remove the `cloned()`. (Note, we had the clone before, but further up)

Reply:
I noticed the move of the clone 😄 

I don't think it's worth sinking much effort into trying to get rid of this clone, and getting the references where they are now is probably faster, I'm not too concerned, just thought it was worth highlighting.

Reply:
I avoided the cloning by conditionally collecting the references before traversing further in cbb81d730b6baa601e9ffd349b7268f55bcf6720. This also allowed me to remove the `Clone` implementation on all records: d8de9653ae6bbcaa1e6c54f34b2622919019d518

Reply:
Nice! Can you add a comment saying why we're collecting them there rather than inline?

Something like "Collecting these here rather than where they're used avoids multiple mutable reference errors" or something

Reply:
> I avoided the cloning by conditionally collecting the references before traversing further in https://github.com/hashintel/hash/commit/cbb81d730b6baa601e9ffd349b7268f55bcf6720. This also allowed me to remove the Clone implementation on all records: https://github.com/hashintel/hash/commit/d8de9653ae6bbcaa1e6c54f34b2622919019d518

Have you done a quick sanity check that we weren't using it in the benchmarks code or anything?

Reply:
I didn't expect that, but the integration tests used an (unneeded) clone: 25a87b200cc95d0b36b43ecaf5a77e464c9cfae8.

655b726b51b6158fc47b33a0d42fca489ec6cbff